### PR TITLE
Do not wrap `otherUI` into credential ID

### LINKED ISSUE #1458: Remove mutable information from credential ID generation
(this is branched from #1456)

We should update generation of [credential IDs that wrap the public key credential source](https://w3c.github.io/webauthn/#credential-id) to remove mutable fields from it (at the moment anything under `otherUI` such as `displayName`, `name`). Credential IDs are immutable, and afaict those fields aren't even used anywhere at the moment.

### PR DESCRIPTION
**Pull request by nsatragno**
Mark `otherUI` as a mutable public key credential source item, and indicate that such items should not be wrapped into credential IDs. This prevents authenticators from binding (mutable) personal information to (immutable) credential IDs, and paves the way to add provisions to update personal information in future versions.

fixes #1458 
improves #1456

***
<a href="https://pr-preview.s3.amazonaws.com/nsatragno/webauthn/pull/1460.html" title="Last updated on Jul 29, 2020, 9:25 PM UTC (d0fecd5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1460/b1c6965...nsatragno:d0fecd5.html" title="Last updated on Jul 29, 2020, 9:25 PM UTC (d0fecd5)">Diff</a>

### CODE REVIEW

**Comment by nsatragno** [comment]
@equalsJeffH please take a look, thanks! ^_^

**Comment by kamolsrisan** [comment]
good

**Comment by kamolsrisan** [comment]
good

**Comment by kamolsrisan** [comment]
good

File: index.bs [hunk 1]
```diff
@@ -1114,7 +1114,8 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 
         :   <dfn>otherUI</dfn>
         ::  OPTIONAL other information used by the [=authenticator=] to inform its UI. For example, this might include the user's
-            {{displayName}}.
+            {{displayName}}. [=otherUI=] is a <dfn>mutable item</dfn> and SHOULD NOT be bound to the [=public key credential source=]
+            in a way that prevents updating it without destroying the [=public key credential source=].
```
Comment (equalsJeffH) [inline index.bs:1118]: how about...
```suggestion
            in a way that prevents its being updated.
```
..?

Comment (nsatragno) [reply index.bs:1118]: I find it hard to parse "its being updated" for some reason. Maybe
```suggestion
            in a way that prevents it from being updated.
```
?

Comment (ve7jtb) [reply index.bs:1118]: I think it still needs the "without destroying the [=public key credential source=]."

Comment (emlun) [reply index.bs]: I feel like this is now a bit ambiguous: does "it" refer to "otherUI" or to "the public key credential source"? Maybe replace `it` with `[=otherUI=]`?

Comment (nsatragno) [reply index.bs]: Done!
